src/network/mod.rs

```diff
@@ -18,19 +18,20 @@
 //! of Bitcoin data and network messages.
 //!
 
-use std::fmt;
-use std::io;
-use std::error;
+use io;
+use core::fmt;
+#[cfg(feature = "std")] use std::error;
 
 pub mod constants;
 
-pub mod address;
-pub use self::address::Address;
-pub mod message;
-pub mod message_blockdata;
-pub mod message_network;
-pub mod message_filter;
-pub mod stream_reader;
+// FIXME: put the networking stuff behind std feature
```

Comment:
Does it never make sense to support network messages without std? Maybe not but if its not a ton of diff why not just support it?

Reply:
The immediate issue I see is that the address module heavily depends on `Ipv?Addr`, especially the `AddrV2` enum, and that has further references in other modules.  I can try to make just those conditional, but I'm not sure about the developer experience of having access to only a somewhat random subset of the protocol objects.  What do you think?

Reply:
Sure, if there's a good reason, just drop the whole thing. I just wanted to make sure we weren't doing it spuriously.

Reply:
To clarify:

- the FIXME was obsolete - the relevant items were already behind an std feature
- almost all of the items that are now gated by std don't actually need to be gated.  instead of let's say `AddrV2` depending directly on `Ipv[46]Addr`, it should depend on a custom object that has gated `into` and `from` conversion functions to the `Ipv[46]Addr` objects.
- but this is an API change and should be in a separate PR